datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -142,8 +148,11 @@ public void send(String destinationUri, DatastreamProducerRecord record, SendCal
           LOG.error(errorMessage, e);
           throw new DatastreamRuntimeException(errorMessage, e);
         }
+        // Update topic-specific metrics and aggregate metrics
+        int numBytes = outgoing.key() != null ? outgoing.key().length : 0 + outgoing.value().length;
```

Comment:
I think you need parentheses to get the right order of operations, otherwise numBytes will just be the key length if non-null and will not include the value length. 

int numBytes = (outgoing.key() != null ? outgoing.key().length : 0) + outgoing.value().length;